core/iwasm/libraries/wasi-nn/include/wasi_nn_types.h

```diff
@@ -48,8 +59,10 @@ typedef struct {
     uint32_t size;
 } tensor_dimensions;
 
+// sync up with
+// https://github.com/WebAssembly/wasi-nn/blob/main/wit/wasi-nn.wit#L27
 // The type of the elements in a tensor.
-typedef enum { fp16 = 0, fp32, up8, ip32 } tensor_type;
+typedef enum { fp16 = 0, fp32, fp64, bf16, u8, i32, i64 } tensor_type;
```

Comment:
I know that it is not a decision from your side, but this is a breaking change. How can we inform the users? Maybe we can add it in the README.

Reply:
May I ask which part is *the breaking change*, `up8` -> `u8`? Or the whole PR?

Reply:
A WASM built before with tensor type `up8` now it will be interpreted as `fp64`. This is a breaking change from wasi-nn, but we might want to inform user to recompile.

Reply:
How about 

```
#if WASM_ENABLE_WASI_EPHEMERAL_NN != 0
typedef enum { fp16 = 0, fp32, fp64, bf16, u8, i32, i64 } tensor_type;
#else
typedef enum { fp16 = 0, fp32, up8, ip32 } tensor_type;
#endif 
```

I assume people only tend to chase the latest wasi-nn spec when using `-DWASM_ENABLE_WASI_EPHEMERAL_NN=1`

Reply:
Makes sense to me! Can we add this decision in the README? :bow:

Reply:
Sure.